OpenNefia.Content/Locale/Funcs/GermanBuiltins.cs

```diff
@@ -0,0 +1,93 @@
+﻿using OpenNefia.Content.DisplayName;
+using OpenNefia.Content.GameObjects;
+using OpenNefia.Core.Game;
+using OpenNefia.Core.GameObjects;
+using OpenNefia.Core.IoC;
+using OpenNefia.Core.Locale;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace OpenNefia.Content.Locale.Funcs
+{
+    [RegisterLocaleFunctions("de_DE")]
+    public static class GermanBuiltins
+    {
+        public const string BaseArticle = "ein";
+        public const string GenderAttributeName = "Gender";
+        public const string PluralAttributeName = "Plural";
+
+        public const string PluralNameAttributeName = "PluralRule";
+        public const string PluralNameAlways = "always";
+
+        public const string GenderNameMale = "male";
+        public const string GenderNameFemale = "female";
+        public const string GenderNameNeutral = "neutral";
+
+        public const string AdjectiveAttributeName = "Adjective";
+        public const string AdjectiveDirectAttributeName = "AdjectiveDirect";
+
+        public const string ArticleSuffixFemale = "e";
+        public const string ArticleSuffixMale = "en";
+        /// <summary>
+        /// Function: name(entity, ignoreSight)
+        /// </summary>
+        /// <hsp>#defcfunc name int tc</hsp>
+        [LocaleFunction("name")]
+        public static string BuiltIn_name(object? obj, string? directness = null)
+        {
+            if (obj is not EntityUid entity)
+            {
+                return Loc.GetString("Elona.GameObjects.Common.Something");
+            }
+
+            var gameSession = IoCManager.Resolve<IGameSessionManager>();
+
+            if (gameSession.IsPlayer(entity))
+                return Loc.GetString("Elona.GameObjects.Common.You");
+
+            var visibilitySys = EntitySystem.Get<VisibilitySystem>();
+
+            if (!visibilitySys.CanSeeEntity(GameSession.Player, entity))
+            {
+                return Loc.GetString("Elona.GameObjects.Common.Something");
+            }
+
+            var name = DisplayNameSystem.GetDisplayName(entity);
+            if (char.IsDigit(name.First()))
+                return name;
+
+            var parts = name.Split(' ');
```

Comment:
Could I understand how the string splitting logic is supposed to work here? Are there extra things like adjectives or special metadata that are being detected in the split string? I ask because some item names are separated by spaces, and would rather move those parts to `_LocData` instead of splitting the final string by hand.

Reply:
the split is to get only the actual noun of the item, in German it's always at the end. the adjectives and other extra words will be in _LocData, in some of the examples it's already so. Adjectives are also modified by gender/referral so it was necessary.

Reply:
Will there ever be any nouns with a space between them?

Reply:
Hmm, it's possible, guess I'll change this

Reply:
is changed in a71227a